### PR TITLE
chore: Bump to 0.86.0; update changelog

### PR DESCRIPTION
**Pull request by weihanglo**
[rendered](https://github.com/weihanglo/cargo/blob/version-bump/CHANGELOG.md)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,32 +1,120 @@
...
+  [#14815](https://github.com/rust-lang/cargo/pull/14815)
+- Several resolver performance enhancements, including:
   - Use `rustc-hash` to speed up resolver
     [#14663](https://github.com/rust-lang/cargo/pull/14663)
   - Share conflict cache between activation retries
     [#14692](https://github.com/rust-lang/cargo/pull/14692)
   - Avoid cloning when iterating using `RcVecIter`
     [#14690](https://github.com/rust-lang/cargo/pull/14690)
+  - Avoid `HttpRegistry` returning `block_until_ready` too early when work is still pending
+    [#14694](https://github.com/rust-lang/cargo/pull/14694)
```
Comment (epage) [inline CHANGELOG.md:100]: There a better way to frame this for users?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,32 +1,120 @@
...
+  [#14736](https://github.com/rust-lang/cargo/pull/14736)
+- Add source replacement info when no matching package found during dependency resolving.
+  [#14715](https://github.com/rust-lang/cargo/pull/14715)
+- Hint for using `crates-io` when `[patch.crates.io]` found.
+  [#14700](https://github.com/rust-lang/cargo/pull/14700)
+- Normalize source paths of Cargo targets for better diagnostics.
+  [#14497](https://github.com/rust-lang/cargo/pull/14497)
+  [#14750](https://github.com/rust-lang/cargo/pull/14750)
+- Docs for optional registry JSON fields.
+- Allow registries to omit empty/default fields in index metadata JSON.
```
Comment (epage) [inline CHANGELOG.md:111]: Is the first one redundant?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,32 +1,120 @@
...
+  [#14747](https://github.com/rust-lang/cargo/pull/14747)
+
 ### Nightly only
 
+- The unstable environment variable `CARGO_RUSTC_CURRENT_DIR` has been removed.
+  [#14799](https://github.com/rust-lang/cargo/pull/14799)
+- 🔥 A new unstable feature flag to generate a JSON schema for the manifest.
+  Useful for external tools to further validate or complete Cargo.toml.
+  ([manifest.schema.json](https://github.com/rust-lang/cargo/blob/master/crates/cargo-util-schemas/manifest.schema.json))
+  [#14683](https://github.com/rust-lang/cargo/pull/14683)
```
Comment (epage) [inline CHANGELOG.md:149]: The way this is written, it makes it sound like this is a feature flag on Cargo when this is a changelog entry for one of our helper libraries.  Do we normally put these here?  

We could frame this as a nightly documentation of a JSON Schema file for manifests

**Review by epage — APPROVED** [review]
I'm fine merging once my comments are resolved one way or the other

File: CHANGELOG.md (hunk 1 above)
Comment (weihanglo) [reply CHANGELOG.md]: I don't think normal users care about details, so I went ahead removing details of these performance improvements.

File: CHANGELOG.md (hunk 3 above)
Comment (weihanglo) [reply CHANGELOG.md]: Thanks for the suggestion. Made it focus on the JSON Schema file only.

**Pull request merged**